### DISCUSSION TITLE
When use alluxio as hive fs,cant get data from alluxio.

### DISCUSSION THREAD

**Discussion by awkwardd** [trinodb/trino#23778] [Q&A] [answered]
Trino version:461 alluxio version:2.9.5
Follow the alluxio tutorial(https://docs.alluxio.io/os/user/stable/en/compute/Trino.html) to create a hive table and save it on alluxio. After adding the fs.alluxio.enabled configuration to the hive catalog, accessing the table can only return metadata but no actual data and there is no error report. Debug the program and find that the AlluxioFileSystem list method do get the file list alluxio:**/**.Why this happen?Is there anything wrong with my operation or configuration?

**Comment by awkwardd** [comment]
@JiamingMai Could you help me plz.

**Reply by JiamingMai** [reply]
@awkwardd Can you see the files on file system correctly when running `./bin/alluxio fs ls /` in Alluxio 2.9.5?

**Reply by awkwardd** [reply]
yes
[hadoop@alluxiomaster alluxio]$ bin/alluxio fs ls /
drwxr-xr-x  hadoop         hadoop                       0       PERSISTED 10-10-2024 22:50:44:867  DIR /.alluxio_ufs_persistencedrwxr-xr-x  hadoop         supergroup                   0       PERSISTED 10-09-2024 02:26:22:158  DIR /alluxio_backups
drwxr-xr-x  hadoop         supergroup                   0       PERSISTED 07-03-2024 23:51:01:515  DIR /data
drwxr-xr-x  hadoop         hadoop                       0       PERSISTED 10-09-2024 23:13:47:481  DIR /default_tests_files
drwxr-xr-x  hadoop         hadoop                       1       PERSISTED 10-10-2024 22:50:44:867  DIR /ml-100k
drwx-wx-wx  hadoop         supergroup                   0       PERSISTED 07-03-2024 23:50:51:385  DIR /tmp 
drwxr-xr-x  hadoop         supergroup                   2       PERSISTED 09-02-2024 02:43:44:636  DIR /user

**Reply by JiamingMai** [reply]
This indicates that Alluxio 2.9.5 can access the underlying file system. Can you share your DDL SQL of the Hive table? So that we will have a test in our environment.

**Reply by awkwardd** [reply]
CREATE TABLE `u_user`(
  `userid` int, 
  `age` int, 
  `gender` char(1), 
  `occupation` string, 
  `zipcode` string)
ROW FORMAT SERDE 
  'org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe' 
WITH SERDEPROPERTIES ( 
  'field.delim'='|', 
  'serialization.format'='|') 
STORED AS INPUTFORMAT 
  'org.apache.hadoop.mapred.TextInputFormat' 
OUTPUTFORMAT 
  'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'
LOCATION
  'alluxio://192.168.88.112:19998/ml-100k'; just as the tutorial say.

**Reply by JiamingMai** [reply]
Thanks. We will have a try.

**Reply by awkwardd** [reply]
Thank you for help me.This has been confusing me for days.

**Comment by awkwardd** [comment]
I find that when use alluxio as fs, the MetaDataManger getTableStatistics() can't return the row count for the query.Is that the reason why the query can't get any data from alluxio?

**Reply by JiamingMai** [reply]
Can you try calling `sync_partition_metadata` and submit the query again?

**Reply by awkwardd** [reply]
It doesn't work.

**Comment by awkwardd** [comment, marked as answer]
I solved this problem by modify the AlluxioFileIterator.java https://github.com/trinodb/trino/pull/23815/commits @JiamingMai .You can take a look.

**Reply by JiamingMai** [reply]
Thanks a lot. We will take a look later.

**Comment by raunaqmorarka** [comment]
Fixed by https://github.com/trinodb/trino/pull/23815
